src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,22 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private final int indexSinceYear = 0;
+    private final int indexByYear = 1;
+    private final int minPeriodsInUkraine = 10;
+    private final int minAge = 35;
+    private final String ukrainianNationality = "Ukrainian";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        String[] split = candidate.getPeriodsInUkr().split("-");
+        return candidate.getAge() >= minAge
+                && candidate.getNationality().equals(ukrainianNationality)
+                && (Integer.parseInt(split[indexByYear])
+                - Integer.parseInt(split[indexSinceYear])) >= minPeriodsInUkraine
```

Comment:
```suggestion
        String[] periodsInUkraine = candidate.getPeriodsInUkr().split(SPLIT_REGEX);
        duration = Integer.parseInt(periodInUkr[YEAR_TO_INDEX])
      - Integer.parseInt(periodInUkr[YEAR_FROM_INDEX]);
        return candidate.getAge() >= MIN_AGE
                && candidate.getNationality().equals(NATIONALITY)
                && duration >= MIN_YEARS_IN_COUNTRY
```
Parsing numbers ouside the boolean expression makes it easier to read